md4c/md4c.c

```diff
@@ -3398,6 +3415,47 @@ md_resolve_links(MD_CTX* ctx, const MD_LINE* lines, int n_lines)
             continue;
         }
 
+        /* Detect and resolve wiki links. */
+        if ((ctx->parser.flags & MD_FLAG_WIKILINKS) &&
+            next_opener != NULL && next_closer != NULL &&
+            (opener->end - opener->beg == 1) &&
+            (next_opener->beg == opener->beg - 1) &&
+            (next_closer->beg == closer->beg + 1) &&
+            (next_opener->end - next_opener->beg == 1) &&
+            (next_closer->end - next_closer->beg == 1) &&
+            (next_opener->ch == '[' && next_closer->ch == ']'))
+        {
+
+            is_link = md_is_inline_wikilink_spec(ctx, lines, opener->end);
+            if (!is_link) {
+                opener_index = next_index;
+                continue;
```

Comment:
You are right about that. Will fix.

Reply:
Although I might add that with the current implementaiton of `md_is_inline_wikilink_spec`, I believe that what you mention is not possible.

Reply:
Please see my newly added comment about `md_resolve_links()`.

This test has to be done differently. You cannot check just the raw text via the `CH(off)` because that `[` might be escaped or already used.

I believe that we have to check `ctx->marks[]` in neighborhood of the opener and closer mark (not sure right now whether its outer or inner side) whether it is also `[` or `]` mark and whether it is not yet "resolved".

Reply:
> Please see my newly added comment about `md_resolve_links()`.
> 
> This test has to be done differently. You cannot check just the raw text via the `CH(off)` because that `[` might be escaped or already used.

Can you think of a test case for this? I am trying to come up with something that would fail in a test, but I have no luck. I think that the giant if-statement that does the wiki link detection prevents such a case from happening.

Reply:
> Can you think of a test case for this?

Took me a while but yes:
```
$ printf '[[ ![foo\]](url) \n]]' | ./md2html/md2html.exe --fwiki-links
<p><x-wikilink data-target=" ![foo]](url)
"> <img src="url" alt="foo]">
</x-wikilink></p>
```

The trick is to use image because image can be nested in a link while link a link cannot be nested. Here your scanner stops on `\\]]` which makes no sense at all. And its real end is on the next line.

Reply:
Well, thinking more about it, it is sort of the same objection as the `md_is_inline_wikilink_spec()` inconsistency. Another counter-example is this:

```
[[`]]`
]]
```

I.e. hiding the false `]]` into a code span as a trap for the `md_is_inline_wikilink_spec()`.

Reply:
With the re-implementation `md_is_inline_wikilink_id`, I get the following from your first test case here:

```
<p>[[ <img src="url" alt="foo]">
]]</p>
```

That looks right, doesn't it? And the second:

```
<p>[[<code>]]</code>
]]</p>
```

Do you think they should be included as test cases?

Reply:
Probably no.